src/mat/interface/permonmatregularize.c

```diff
@@ -208,6 +208,7 @@ PetscErrorCode MatRegularize(Mat K, Mat R, MatRegularizationType type, Mat *newK
   Mat               Q_loc, Kreg;
   PetscScalar       rho;
   Mat               K_loc, R_loc;
+  PetscInt          regularized_int;
```

Comment:
```suggestion
  int               regularized_int;
```
as PetscObjectComposedDataGetInt() works with int

Reply:
Also please fix `regularized_id` above. My fault.

Reply:
I just checked, the `PetscInt regularized_int` is correct (documentation is wrong). The `regularized_id` is of type `PetscObjectState` which is `PetscInt64`. I'm going to change the id to the correct type and declare the `regularized_int` as `PETSC_UNUSED` to get rid of the `(void)` cast below.

Reply:
`regularized_id` will need to be `PetscInt` due to `PetscObjectComposedDataRegister()`

Reply:
Yes, you're right, I was checking just the manpage - should be fixed in PETSc, I will do it.